From bfc49a7635f020b7375e6dccfc46bc716134a39b Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Wed, 1 Jun 2022 16:14:18 +0800
Subject: [PATCH 30/31] Revert "drm/bridge: Fix free wrong object in
 sii8620_init_rcp_input_dev"

This reverts commit e7a0c8546f74113ea69391b04ac47eea7e380c03.
---
 drivers/gpu/drm/bridge/sil-sii8620.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/sil-sii8620.c b/drivers/gpu/drm/bridge/sil-sii8620.c
index fb0b64c965b7..04431dbac4a4 100644
--- a/drivers/gpu/drm/bridge/sil-sii8620.c
+++ b/drivers/gpu/drm/bridge/sil-sii8620.c
@@ -2118,7 +2118,7 @@ static void sii8620_init_rcp_input_dev(struct sii8620 *ctx)
 	if (ret) {
 		dev_err(ctx->dev, "Failed to register RC device\n");
 		ctx->error = ret;
-		rc_free_device(rc_dev);
+		rc_free_device(ctx->rc_dev);
 		return;
 	}
 	ctx->rc_dev = rc_dev;
-- 
2.27.0

